### PR TITLE
Add EndOfFrame implementation using `UnityEngine.Awaitable`

### PR DESCRIPTION
**Pull request by hadashiA**
Currently, MonoBehaviour is required to implement EndOfFrame, but Awaitable.EndOfFrameAsync is available from Unity 2023.1.

### CODE REVIEW

File: src/UniTask/Assets/Plugins/UniTask/Runtime/UniTask.Delay.cs [hunk 1]
```diff
@@ -75,6 +75,12 @@ public static UniTask NextFrame(PlayerLoopTiming timing, CancellationToken cance
             return new UniTask(NextFramePromise.Create(timing, cancellationToken, out var token), token);
         }
 
+#if UNITY_2023_1_OR_NEWER
+        public static async UniTask WaitForEndOfFrame(CancellationToken cancellationToken = default)
```
Comment (hadashiA) [inline src/UniTask/Assets/Plugins/UniTask/Runtime/UniTask.Delay.cs:79]: I have done a easy implementation that uses `async`.
Should I consider adding Awaitable.ToUniTask()?

Comment (neuecc) [reply src/UniTask/Assets/Plugins/UniTask/Runtime/UniTask.Delay.cs:79]: OK to use async.

**Comment by neuecc** [comment]
EndOfFrame thing is mentioned a lot in the ReadMe, I'd like to see an addendum in the ReadMe about the behavior in Unity 2023.1 and later.

**Comment by hadashiA** [comment]
I've updated the README !

**Comment by neuecc** [comment]
thanks!

**Pull request merged**